flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/DefaultResourceAllocationStrategy.java

```diff
@@ -135,7 +154,45 @@ private static List<InternalResourceInfo> getPendingResources(
                 .collect(Collectors.toList());
     }
 
-    private static int tryFulfilledRequirementWithResource(
+    private int tryFulfilledRequirementWithAvailableResources(
```

Comment:
I think there's not much differences between fulfilling requirements from existing available resources and pending resources. We only need an argument to tell which strategy should be used.